File: knowledge/src/main/java/com/ayvytr/knowledge/presenter/GankHistoryPresenter.java
package com.ayvytr.knowledge.presenter;

import com.ayvytr.commonlibrary.BaseObserver;
import com.ayvytr.commonlibrary.bean.GankHistory;
import com.ayvytr.knowledge.R;
import com.ayvytr.knowledge.contract.GankHistoryContract;
import com.ayvytr.knowledge.model.GankHistoryModel;
import com.ayvytr.mvpbase.BasePresenter;
import com.ayvytr.mvprxlifecycle.RxUtils;

/**
 * @author admin
 */
public class GankHistoryPresenter extends BasePresenter<GankHistoryContract.Model, GankHistoryContract.View> {
    public GankHistoryPresenter(GankHistoryContract.View view) {
        super(new GankHistoryModel(), view);
    }

    public void requestGankHistory() {
        mModel.getGankHistory()
              .compose(RxUtils.<GankHistory>bindToLifecycle(mView))
              .compose(RxUtils.<GankHistory>subscribeIo(mView))
              .subscribe(new BaseObserver<GankHistory>() {
                  @Override
                  public void onNext(GankHistory gankHistory) {
                      if(gankHistory.isError()) {
                          mView.showError(R.string.request_data_error);
                      } else {
                         mView.showGankHistory(gankHistory.getResults());
                      }
                  }
              });
    }
}

File: knowledge/src/main/java/com/ayvytr/knowledge/presenter/GankHistoryPresenter.kt
package com.ayvytr.knowledge.presenter

import com.ayvytr.commonlibrary.BaseObserver
import com.ayvytr.commonlibrary.bean.GankHistory
import com.ayvytr.knowledge.R
import com.ayvytr.knowledge.contract.GankHistoryContract
import com.ayvytr.knowledge.model.GankHistoryModel
import com.ayvytr.mvp.BasePresenter
import com.ayvytr.rxlifecycle.RxUtils

/**
 * @author admin
 */
class GankHistoryPresenter(view: GankHistoryContract.View) :
    BasePresenter<GankHistoryContract.Model, GankHistoryContract.View>(GankHistoryModel(), view) {

    fun requestGankHistory() {
        mModel.gankHistory
            .compose(RxUtils.bindToLifecycle(mView))
            .compose(RxUtils.ofDefault(mView))
            .subscribe(object : BaseObserver<GankHistory>() {
                override fun onNext(gankHistory: GankHistory) {
                    if (gankHistory.isError) {
                        mView.showError(R.string.request_data_error)
                    } else {
                        mView.showGankHistory(gankHistory.results!!)
                    }
                }
            })
    }
}
